rewrite-properties/src/main/java/org/openrewrite/properties/AddProperty.java

```diff
@@ -41,13 +44,37 @@ public class AddProperty extends Recipe {
             description = "The value of the new property key.")
     String value;
 
+    @Option(displayName = "Optional comment to be prepended to the property",
+            description = "A comment that will be added to the new property.",
+            required = false,
+            example = "This is a comment")
+    @Nullable
+    String comment;
+
     @Option(displayName = "Optional delimiter",
             description = "Property entries support different delimiters (`=`, `:`, or whitespace). The default value is `=` unless provided the delimiter of the new property entry.",
             required = false,
             example = ":")
     @Nullable
     String delimiter;
 
+    /**
+     * Keeping this constructor just for compatibility purposes
+     * @deprecated Use {@link AddProperty#AddProperty(String, String, String, String)}}
+     */
+    @Deprecated
+    public AddProperty(String property, String value, @Nullable String delimiter) {
+        this(property, value, null, delimiter);
+    }
+
+    @JsonCreator
```

Comment:
I could even lean towards removing this old constructor and making the change downstream if needed (which I'm open to do). That way there's no confusion going forward, especially as the three and four argument constructors have a different third argument, which could be confusing.
```suggestion
```